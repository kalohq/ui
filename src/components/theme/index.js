/* @flow */
export {default} from './theme';
