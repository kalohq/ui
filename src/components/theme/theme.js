/* @flow */

import tokens from '../../design-tokens/tokens.theme.js';

/**
 * ThemeProvider Kalo Theme
 * 
 * This is the core CSS-in-JS theme that is passed in to the
 * global ThemeProvider. It is mainly composed of our core
 * design tokens.
 */

const theme = {
  space: [0, 8, 16, 32, 64],
  breakpoints: [],
  ...tokens,
};

export default theme;
