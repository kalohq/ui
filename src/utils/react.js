import ReactDOM from 'react-dom';

/**
 * Focus a ref’d react input when it is mounted
 * @param ref
 */
export function focusOnMount(ref) {
  const el = ReactDOM.findDOMNode(ref);

  if (el) {
    // Give focus and put the cursor at the end of
    // the value.
    el.focus();
    el.value = el.value;
  }
}

/**
 * Select the text of a ref’d react input when it is mounted
 * @param ref
 */
export function selectOnMount(ref) {
  if (ref && ref.select) {
    ref.select();
  }
}
