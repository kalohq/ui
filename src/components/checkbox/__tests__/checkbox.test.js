/* @flow */
/* eslint-env jest */
import * as React from 'react';
import * as emotion from 'emotion';
import {createSerializer} from 'jest-emotion';
import renderer from 'react-test-renderer';
import {ThemeProvider} from 'emotion-theming';

import theme from 'components/theme';
import Checkbox from 'components/checkbox';

expect.addSnapshotSerializer(createSerializer(emotion));

describe('Checkbox', () => {
  const defaultProps = {
    size: 'medium',
    checked: true,
  };
  const create = (props = {}) =>
    renderer
      .create(
        <ThemeProvider theme={theme}>
          <Checkbox {...defaultProps} {...props} />
        </ThemeProvider>
      )
      .toJSON();

  test('should render shallow component ok', () => {
    const element = create();
    expect(element).toMatchSnapshot();
  });

  test('should render a checkbox with label', () => {
    const element = create({
      label: 'A checkbox label',
    });

    expect(element).toMatchSnapshot();
  });

  test('should render a checkbox partially checked', () => {
    const element = create({
      label: 'A checkbox label',
      checked: false,
      indeterminate: true,
    });
    expect(element).toMatchSnapshot();
  });
});
