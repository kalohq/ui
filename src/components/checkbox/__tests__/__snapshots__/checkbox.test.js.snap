// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render a checkbox partially checked 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
}

.emotion-3 {
  display: inline-block;
  vertical-align: bottom;
  margin-left: 10px;
  font-family: 'WebFaktSoftPro',sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  cursor: inherit;
  text-align: unset;
  text-decoration: inherit;
  font-weight: 400;
  font-size: 14px;
  color: rgb(84,103,137);
  white-space: normal;
  text-overflow: initial;
  overflow: visible;
  white-space: pre-wrap;
  word-break: break-word;
}

.emotion-3:hover {
  text-decoration: inherit;
}

.emotion-3 strong {
  font-weight: 500;
}

.emotion-1 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  border-radius: 3px;
  border: 1px solid rgb(176,186,195);
  -webkit-transition: 100ms ease-in-out background-color,border-color;
  transition: 100ms ease-in-out background-color,border-color;
  height: 14px;
  width: 14px;
  background-size: cover;
  background-position: center center;
  background-repeat: no-repeat;
  background-color: rgb(255,255,255);
  background-color: rgb(249,250,252);
  background-image: url(path/to/file);
  border: 1px solid rgb(84,195,219);
}

<div
  aria-checked="mixed"
  className="emotion-4 emotion-5"
  disabled={false}
  name={undefined}
  onClick={undefined}
  role="checkbox"
>
  <div
    checked={false}
    className="emotion-0 emotion-1"
    disabled={false}
    size="medium"
    style={Object {}}
  />
  <span
    className="emotion-2 emotion-3"
    color="navy600"
    onClick={undefined}
    size="small"
    style={Object {}}
  >
    A checkbox label
  </span>
</div>
`;

exports[`Checkbox should render a checkbox with label 1`] = `
.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
}

.emotion-1 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  border-radius: 3px;
  border: 1px solid rgb(176,186,195);
  -webkit-transition: 100ms ease-in-out background-color,border-color;
  transition: 100ms ease-in-out background-color,border-color;
  height: 14px;
  width: 14px;
  background-size: cover;
  background-position: center center;
  background-repeat: no-repeat;
  background-color: rgb(255,255,255);
  background-color: rgb(249,250,252);
  background-image: url(path/to/file);
  border: 1px solid rgb(84,195,219);
}

.emotion-3 {
  display: inline-block;
  vertical-align: bottom;
  margin-left: 10px;
  font-family: 'WebFaktSoftPro',sans-serif;
  white-space: nowrap;
  text-overflow: ellipsis;
  overflow: hidden;
  cursor: inherit;
  text-align: unset;
  text-decoration: inherit;
  font-weight: 400;
  font-size: 14px;
  color: rgb(84,103,137);
  white-space: normal;
  text-overflow: initial;
  overflow: visible;
  white-space: pre-wrap;
  word-break: break-word;
}

.emotion-3:hover {
  text-decoration: inherit;
}

.emotion-3 strong {
  font-weight: 500;
}

<div
  aria-checked={true}
  className="emotion-4 emotion-5"
  disabled={false}
  name={undefined}
  onClick={undefined}
  role="checkbox"
>
  <div
    checked={true}
    className="emotion-0 emotion-1"
    disabled={false}
    size="medium"
    style={Object {}}
  />
  <span
    className="emotion-2 emotion-3"
    color="navy600"
    onClick={undefined}
    size="small"
    style={Object {}}
  >
    A checkbox label
  </span>
</div>
`;

exports[`Checkbox should render shallow component ok 1`] = `
.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  cursor: pointer;
}

.emotion-1 {
  position: relative;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  border-radius: 3px;
  border: 1px solid rgb(176,186,195);
  -webkit-transition: 100ms ease-in-out background-color,border-color;
  transition: 100ms ease-in-out background-color,border-color;
  height: 14px;
  width: 14px;
  background-size: cover;
  background-position: center center;
  background-repeat: no-repeat;
  background-color: rgb(255,255,255);
  background-color: rgb(249,250,252);
  background-image: url(path/to/file);
  border: 1px solid rgb(84,195,219);
}

<div
  aria-checked={true}
  className="emotion-2 emotion-3"
  disabled={false}
  name={undefined}
  onClick={undefined}
  role="checkbox"
>
  <div
    checked={true}
    className="emotion-0 emotion-1"
    disabled={false}
    size="medium"
    style={Object {}}
  />
</div>
`;
